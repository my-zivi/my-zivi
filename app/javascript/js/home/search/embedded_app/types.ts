/* eslint-disable camelcase */

export interface JobPostingSearchHit {
  objectID: string;
  title: string;
  icon_url: string;
  publication_date: string;
  plain_description: string;
  description: string;
  brief_description: string;
  required_skills?: string;
  preferred_skills?: string;
  canton: string;
  identification_number: string;
  relevancy: Relevancy;
  contact_information: string;
  minimum_service_months: number;
  language: string;
  organization_display_name: string;
  category_display_name: string;
  sub_category_display_name?: string;
  link: string;
  featured_as_new: boolean;
  priority_program: boolean;
}

export enum Relevancy {
  Normal = 1,
  Urgent,
  Featured
}
