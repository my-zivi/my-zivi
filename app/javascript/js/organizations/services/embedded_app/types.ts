export interface Service {
  beginning: string;
  ending: string;
  civilServant: {
    fullName: string;
  };
}
