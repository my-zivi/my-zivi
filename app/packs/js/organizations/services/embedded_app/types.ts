export interface Service {
  beginning: string;
  ending: string;
  definitive: boolean;
  civilServant: {
    fullName: string;
  };
}
