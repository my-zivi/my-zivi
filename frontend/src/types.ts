import { FormikBag } from 'formik';

export interface Holiday {
  id?: number;
  beginning: string;
  ending: string;
  holiday_type: 'public_holiday' | 'company_holiday';
  description: string;
}

export enum PaymentState {
  payment_in_progress = 'payment_in_progress',
  paid = 'paid',
}

export interface Payment {
  id?: number;
  total: number;
  state: PaymentState;
  payment_timestamp: number;
  expense_sheets: PaymentExpenseSheet[];
}

export interface PaymentExpenseSheet {
  id?: number;
  total: number;
  user: {
    id: number;
    bank_iban: string;
    full_name: string;
    zdp: number;
  };
}

export interface ExpenseSheet {
  id: number;
  bank_account_number: string;
  beginning: Date;
  clothing_expenses: number;
  clothing_expenses_comment: null;
  company_holiday_comment: null;
  driving_expenses: number;
  driving_expenses_comment: null;
  duration: number;
  ending: Date;
  extraordinary_expenses: number;
  extraordinary_expenses_comment: null;
  paid_company_holiday_days: number;
  paid_vacation_comment: null;
  paid_vacation_days: number;
  payment_timestamp: null;
  sick_comment: null;
  sick_days: number;
  state: ExpenseSheetState;
  total: number;
  unpaid_company_holiday_days: number;
  unpaid_vacation_comment: null;
  unpaid_vacation_days: number;
  user_id: number;
  work_days: number;
  workfree_days: number;
  service_id: number;
}

export interface ExpenseSheetHints {
  suggestions: {
    work_days: number;
    workfree_days: number;
    paid_company_holiday_days: number;
    unpaid_company_holiday_days: number;
    clothing_expenses: number;
  };
  remaining_days: {
    sick_days: number;
    paid_vacation_days: number;
  };
}

export enum ExpenseSheetState {
  open = 'open',
  ready_for_payment = 'ready_for_payment',
  payment_in_progress = 'payment_in_progress',
  paid = 'paid',
}

export interface ShortExpenseSheetListing {
  id: number;
  ending: string;
  beginning: string;
  state: ExpenseSheetState;
  duration: number;
}

export interface ExpenseSheetListing extends ShortExpenseSheetListing {
  total: number;
  user: {
    id: number;
    address: string;
    bank_iban: string;
    city: string;
    full_name: string;
    zdp: number;
    zip: number;
  };
}

export interface DailyExpense {
  breakfast: number;
  lunch: number;
  dinner: number;
}

export interface ServiceSpecification {
  id: number;
  identification_number?: number;
  name: string;
  short_name: string;
  work_clothing_expenses: number;
  work_days_expenses: DailyExpense;
  paid_vacation_expenses: DailyExpense;
  first_day_expenses: DailyExpense;
  last_day_expenses: DailyExpense;
  accommodation_expenses: number;
  pocket_money: number;
  active: boolean;
}

export interface User {
  id: number;
  active: boolean;
  address: string;
  bank_iban: string;
  birthday: string;
  chainsaw_workshop: boolean;
  city: string;
  driving_licence_b: boolean;
  driving_licence_be: boolean;
  email: string;
  ending: null | string;
  first_name: string;
  health_insurance: string;
  hometown: string;
  internal_note: string;
  last_name: string;
  phone: string;
  regional_center_id: number;
  expense_sheets: ShortExpenseSheetListing[];
  role: 'admin' | 'civil_servant';
  services: Service[];
  beginning: null | string;
  work_experience: null | string;
  zdp: number;
  zip: number | null;
}

export interface RegionalCenter {
  name: string;
  address: string;
  short_name: string;
  id: number;
}

export interface UserFilter {
  zdp: string;
  name: string;
  beginning: string;
  ending: string;
  active: boolean;
  role: string;
}

export interface Service {
  id?: number;
  beginning: Date | null;
  service_days: number;
  confirmation_date: null | string;
  eligible_paid_vacation_days: number;
  ending: Date | null;
  first_swo_service: boolean;
  long_service: boolean;
  service_type: string | null;
  probation_period: boolean;
  service_specification_id: number;
  service_specification: {
    identification_number: number;
    name: string | undefined;
    short_name: string | undefined;
  };
  user_id: number;
}

export interface ServiceCollection {
  id?: number;
  beginning: string | null;
  ending: string | null;
  confirmation_date: string | null;
  service_specification: {
    identification_number: number;
    name: string | undefined;
    short_name: string | undefined;
  };
  user: {
    id: number;
    first_name: string | null;
    last_name: string | null;
    zdp: number;
  };
}

export interface UserFeedback {
  id?: number;
}

export interface UserQuestionWithAnswers {
  id?: number;
  answers: UserQuestionAnswers;
  custom_info: string;
  opt1: string;
  opt2: string;
  question: string;
  type: number;
}

export interface UserQuestionAnswers {
  1: number;
  2: number;
  3: number;
  4: number;
  5: number;
  6: number;
}

export interface Listing {
  id?: number;
  archived?: boolean;
}

export interface Column<T> {
  id: string;
  numeric?: boolean;
  label: string;
  format?: (t: T) => React.ReactNode;
  span?: {
    col?: number;
    row?: number;
  };
}

export type ActionButtonAction = (() => void) | string;

// tslint:disable-next-line:no-any ; really don't care for that type, and it comes from deep inside Formik
export type HandleFormikSubmit<Values> = (values: Values, formikBag: FormikBag<any, Values>) => void;

// If we'd type thoroughly we'd need to create a type for each models representation in a form / yup validation schema
// tslint:disable-next-line:no-any
export type FormValues = any;
