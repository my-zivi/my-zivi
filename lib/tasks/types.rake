# frozen_string_literal: true

def js_type(indentation, value)
  case value
  when String
    'string'
  when Array
    'string[]'
  else
    to_js_object(value, indentation + 2)
  end
end

def to_js_object(hash, indentation = 2)
  content = hash.map do |key, value|
    indent = ' ' * indentation

    "#{indent}#{key}: #{js_type(indentation, value)};"
  end

  base_indent = ' ' * (indentation - 2)

  "{\n#{content.join("\n")}\n#{base_indent}}"
end

namespace :types do
  desc 'Generates Typescript translation types'
  task generate: :environment do
    content = to_js_object(I18n.t('search').deep_transform_keys { |key| key.to_s.camelize(:lower) })

    template = <<~JS
      // Generated file using `rake types:generate`

      declare interface SearchTranslations #{content[..-1]}
    JS

    File.write(Rails.root.join('app/javascript/js/shared/SearchTranslations.d.ts'), template)
  end
end
