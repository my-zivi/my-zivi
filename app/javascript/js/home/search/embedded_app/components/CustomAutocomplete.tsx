import * as React from 'preact/compat';
import { AutocompleteProvided, connectAutoComplete } from 'react-instantsearch-core';
import { JobPostingSearchHit } from 'js/home/search/embedded_app/types';
import {
  isEmpty, isInteger, take, uniqBy,
} from 'lodash';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import SearchBox, { SearchBoxRef } from './SearchBox';

type Props = AutocompleteProvided<JobPostingSearchHit>;
type State = {
  focused: boolean,
  isHovering: boolean,
  selectedHitIndex?: number,
};

export class AutocompleteImpl extends React.Component<Props, State> {
  private readonly searchBoxRef;
  private flattenedHits: JobPostingSearchHit[] = [];

  constructor(props: Props) {
    super(props);
    this.searchBoxRef = React.createRef<SearchBoxRef>();
    this.state = {
      focused: false,
      isHovering: false,
      selectedHitIndex: undefined,
    };
  }

  componentDidMount(): void {
    document.addEventListener('keyup', this.onKeyup);
  }

  componentWillUnmount(): void {
    document.removeEventListener('keyup', this.onKeyup);
  }

  componentWillUpdate(nextProps: Readonly<Props>): void {
    this.flattenedHits = take(uniqBy(nextProps.hits, (hit) => hit.title), 10);
  }

  render(): JSX.Element {
    return (
      <>
        <SearchBox
          ref={this.searchBoxRef}
          currentRefinement={this.props.currentRefinement}
          refine={this.props.refine}
          onFocus={() => this.showAutocomplete()}
          onBlur={() => this.collapseAutocomplete()}
        />
        <div className={this.autocompleteContainerClass()}>
          <div
            className="autocomplete-content"
            onMouseEnter={() => this.setState({ isHovering: true })}
            onTouchStart={() => this.setState({ isHovering: true })}
            onTouchEnd={() => this.setState({ isHovering: false })}
            onMouseLeave={() => this.setState({ isHovering: false })}
          >
            {this.renderHitsList()}
          </div>
        </div>
      </>
    );
  }

  private onKeyup = (event: KeyboardEvent) => {
    let { selectedHitIndex } = this.state;

    switch (event.key.toLocaleLowerCase()) {
      case 'arrowup':
        selectedHitIndex = (selectedHitIndex || this.flattenedHits.length) - 1;
        break;
      case 'arrowdown':
        selectedHitIndex = !isInteger(selectedHitIndex) ? 0 : selectedHitIndex + 1;
        break;
      case 'enter':
        this.onAutocompleteHitClick(this.flattenedHits[selectedHitIndex]);
        return;
      default:
        break;
    }

    selectedHitIndex %= this.flattenedHits.length;
    this.setState({ selectedHitIndex });
  };

  private collapseAutocomplete() {
    this.setState({ focused: false });
  }

  private showAutocomplete() {
    this.setState({ focused: true, selectedHitIndex: undefined });
  }

  private renderHitsList() {
    return (
      <ul>
        {this.flattenedHits.map((hit, index) => (
          <a
            href="#"
            key={hit.objectID}
            onClick={(e) => this.onAutocompleteHitClick(hit, e)}
            className="autocomplete-hit"
          >
            <li
              className={this.state.selectedHitIndex === index ? 'selected' : undefined}
              onMouseOver={() => this.setState({ selectedHitIndex: index })}
            >
              <FontAwesomeIcon icon={['fas', 'search']} className="mr-2" />
              {hit.title}
            </li>
          </a>
        ))}
      </ul>
    );
  }

  private onAutocompleteHitClick(hit?: JobPostingSearchHit, event?: MouseEvent) {
    event?.preventDefault();
    if (!hit) {
      return;
    }

    this.searchBoxRef.current.autocompleteSearch(hit.title);
    this.setState({ isHovering: false, focused: false, selectedHitIndex: undefined });
  }

  private autocompleteContainerClass() {
    const canPresent = this.state.focused || this.state.isHovering;
    const isVisible = canPresent && !isEmpty(this.props.currentRefinement) && !isEmpty(this.props.hits);
    return `autocomplete-container ${isVisible ? 'd-block' : 'd-none'}`;
  }
}

export default connectAutoComplete(AutocompleteImpl as never);
