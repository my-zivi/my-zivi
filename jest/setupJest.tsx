/* eslint-disable @typescript-eslint/no-explicit-any */
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-preact-pure';
import React, { h, Fragment } from 'preact';
import '../app/javascript/js/shared/globals.d';

(global as any).h = h;
(global as any).Fragment = Fragment;

const globalConfig: MyZiviGlobal = {
  locale: 'de',
  translations: {
    servicesOverview: {
      title: 'My Services Overview',
      errors: {
        cannotLoad: 'Konnte die EinsÃ¤tze nicht laden',
      },
    },
    search: {
      loadMore: 'string',
      noResultsFound: 'string',
      poweredBy: 'string',
      refinements: {
        canton: 'string',
        category: 'string',
        collapseButton: 'string',
        controls: {
          noResults: 'string',
          placeholder: 'string',
          reset: 'string',
          resetTitle: 'string',
          showLess: 'string',
          showMore: 'string',
          submitTitle: 'string',
        },
        language: 'string',
        languages: {
          french: 'string',
          german: 'string',
          italian: 'string',
        },
        month: {
          one: 'string',
          other: 'string',
        },
        minimumServiceMonths: 'string',
        priorityProgram: 'string',
        priorityProgramLabel: 'string',
      },
      ribbons: {
        featured: 'Feat.',
        featuredFull: 'Featured',
        new: 'Neu',
        urgent: 'Wichtig',
      },
      search: 'string',
      searchPlaceholder: 'string',
      statistics: 'string',
      searchView: {
        map: 'Karte',
        tiles: 'Raster',
      },
    } as any,
    loading: 'Laden...',
    error: 'Fehler',
  },
  paths: {
    servicesOverview: '/my/mock/services.json',
  },
  algolia: {
    apiKey: 'my-api-key',
    applicationId: 'my-application-id',
  },
};
(global as any).MyZivi = globalConfig;

jest.mock('@rails/ujs', () => jest.requireActual('js/shared/__mocks__/Rails'));

jest.mock('@fortawesome/react-fontawesome', () => {
  const component: React.FunctionComponent<any> = () => <></>;
  component.displayName = 'FontAwesomeIcon';
  return {
    FontAwesomeIcon: component,
  };
});

configure({ adapter: new Adapter() });
