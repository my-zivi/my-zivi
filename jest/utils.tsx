/* eslint-disable @typescript-eslint/no-explicit-any */
import { mount } from 'enzyme';
import { IApi } from 'js/shared/Api';
import preact from 'preact';
import { ApiContext } from 'js/shared/ApiProvider';
import React from 'preact/compat';
import { InstantSearch } from 'react-instantsearch-dom';
import { defer } from 'lodash';
import { mockClient } from '../app/javascript/js/tests/algolia/mocks';

type MountComponentType = (componentInstance: preact.JSX.Element, targetComponentName?: string) => any;
export const mountConsumer: MountComponentType = (componentInstance, targetComponentName) => {
  const root = mount(componentInstance);

  if (!targetComponentName) {
    return root;
  }

  return root.find(targetComponentName);
};

type FakeWrapInApiProviderType = (api: IApi) => (wrapped: preact.JSX.Element, component: string) => any;
export const fakeWrapInApiProvider: FakeWrapInApiProviderType = (api: IApi) => (
  (wrapped, component) => mountConsumer(
    (
      <ApiContext.Provider value={api}>
        {wrapped}
      </ApiContext.Provider>
    ),
    component,
  )
);

export const mountInstantSearchClient: MountComponentType = (componentInstance, targetComponentName?) => mountConsumer(
  <InstantSearch searchClient={mockClient} indexName="JobPosting">
    {componentInstance}
  </InstantSearch>,
  targetComponentName,
);

type DeferredCheck = (rendered: { update: () => void }, callback: () => void | Promise<void>) => Promise<void>;
export const deferredCheck: DeferredCheck = (rendered, callback) => new Promise<void>((resolve, reject) => {
  defer(() => {
    rendered.update();
    const result = callback();
    if (!result) {
      return resolve();
    }

    return result.then(resolve).catch(reject);
  });
});
