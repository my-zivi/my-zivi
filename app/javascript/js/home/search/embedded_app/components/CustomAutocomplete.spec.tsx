import React from 'preact/compat';
import { Hit } from 'react-instantsearch-core';
import { mount } from 'enzyme';
import { deferredCheck, mountInstantSearchClient } from '../../../../../../../jest/utils';
import CustomAutocomplete, { AutocompleteImpl } from './CustomAutocomplete';
import JobPostingFactory from '../../../../tests/factories/JobPostingFactory';
import { JobPostingSearchHit } from '../types';

describe('CustomAutocomplete', () => {
  let root;

  beforeEach(() => {
    root = mountInstantSearchClient(<CustomAutocomplete defaultRefinement="alp" />);
  });

  it('renders correctly', () => deferredCheck(root, () => {
    const rendered = root.find('AutocompleteImpl');
    expect(rendered).toMatchSnapshot();
  }));

  it('renders an autocomplete menu when focusing input element', () => deferredCheck(root, () => {
    const inputElement = root.find('input[type="search"]');
    inputElement.simulate('focus');

    expect(root.find('AutocompleteImpl')).toMatchSnapshot();
  }));

  it('collapses autocomplete when mouse moves out and input blurred', () => deferredCheck(root, () => {
    const inputElement = root.find('input[type="search"]');
    inputElement.simulate('focus');
    root.find('.autocomplete-content').simulate('mouseenter');

    const autocompleteContainerClassName = () => root.find('.autocomplete-container').props().className;
    expect(autocompleteContainerClassName()).toMatch(/d-block/);

    inputElement.simulate('blur');
    expect(autocompleteContainerClassName()).toMatch(/d-block/);

    root.find('.autocomplete-content').simulate('mouseleave');
    expect(autocompleteContainerClassName()).toMatch(/d-none/);
  }));

  it('collapses autocomplete touch mouse ends and input blurred', () => deferredCheck(root, () => {
    const inputElement = root.find('input[type="search"]');
    inputElement.simulate('focus');
    root.find('.autocomplete-content').simulate('touchstart');
    expect(root.find('.autocomplete-container').props().className).toMatch(/d-block/);

    inputElement.simulate('blur');
    expect(root.find('.autocomplete-container').props().className).toMatch(/d-block/);

    root.find('.autocomplete-content').simulate('touchend');
    expect(root.find('.autocomplete-container').props().className).toMatch(/d-none/);
  }));

  describe('when a user clicks on an autocomplete item', () => {
    const refine = jest.fn();

    it('searches for the given term', () => {
      const hits = [{
        objectID: '123',
        ...JobPostingFactory.build({ title: 'Mitarbeiter Jugendhilfe' }),
      } as Hit<JobPostingSearchHit>];
      root = mount(<AutocompleteImpl refine={refine} hits={hits} currentRefinement="" />);
      root.setProps({ currentRefinement: 'Mitarbeiter' });

      root.find('.autocomplete-content').simulate('mouseenter');
      root.find('.autocomplete-hit').simulate('click');
      expect(refine).toHaveBeenCalledWith('Mitarbeiter Jugendhilfe');
    });
  });

  describe('when the user presses the enter key', () => {
    const refine = jest.fn();

    it('searches for the selected term', () => deferredCheck(root, () => {
      const hits = [{
        objectID: '123',
        ...JobPostingFactory.build({ title: 'Mitarbeiter Jugendhilfe' }),
      } as Hit<JobPostingSearchHit>];
      root = mount(<AutocompleteImpl refine={refine} hits={hits} currentRefinement="" />);
      root.setProps({ currentRefinement: 'Mitarbeiter' });
      root.update();

      document.dispatchEvent(new KeyboardEvent('keyup', { key: 'arrowdown', keyCode: 40 }));
      document.dispatchEvent(new KeyboardEvent('keyup', { key: 'enter', keyCode: 13 }));
      root.update();

      expect(refine).toHaveBeenCalledWith('Mitarbeiter Jugendhilfe');
    }));
  });
});
