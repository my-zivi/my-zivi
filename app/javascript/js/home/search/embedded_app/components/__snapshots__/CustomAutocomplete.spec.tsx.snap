// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAutocomplete renders an autocomplete menu when focusing input element 1`] = `
<AutocompleteImpl
  createURL={[Function]}
  currentRefinement="alp"
  defaultRefinement="alp"
  hits={
    Array [
      Object {
        "__position": 1,
        "_highlightResult": Object {
          "organization_display_name": Object {
            "fullyHighlighted": false,
            "matchLevel": "full",
            "matchedWords": Array [
              "al",
            ],
            "value": "<ais-highlight-0000000000>Al</ais-highlight-0000000000>",
          },
          "plain_description": Object {
            "fullyHighlighted": false,
            "matchLevel": "full",
            "matchedWords": Array [
              "al",
            ],
            "value": "100% Dauer <ais-highlight-0000000000>Al</ais-highlight-0000000000>psommer",
          },
          "title": Object {
            "fullyHighlighted": false,
            "matchLevel": "full",
            "matchedWords": Array [
              "al",
            ],
            "value": "<ais-highlight-0000000000>Al</ais-highlight-0000000000>p-Pflege",
          },
        },
        "brief_description": "Bereich Kulturlandschaft, Biodiversität und Landschaftsqualität",
        "canton_display_name": "Obwalden",
        "category_display_name": "Landwirtschaft",
        "contact_information": "Dieser Betrieb ist noch nicht bei MyZivi registriert. Bitte bewerbe Dich im EZIVI.",
        "description": "MyDescription <b>Using HTML</b>",
        "featured_as_new": true,
        "icon_url": "/myzivi-logo.jpg",
        "identification_number": 89372,
        "language": "german",
        "link": "http://localhost:3000/job_postings/3",
        "minimum_service_months": 1,
        "objectID": "3",
        "organization_display_name": "Alpbetrieb Untersteiglen-Schwand im Tal-Melchsee-Frutt",
        "plain_description": "100% Dauer Alpsommer: von 01.05. bis 15.10",
        "preferred_skills": null,
        "priority_program": true,
        "publication_date": "2021-06-14",
        "published": true,
        "relevancy": 1,
        "required_skills": "<p> Motivation und physische Kondition für harte körperliche Arbeit im Freien</p>",
        "sub_category_display_name": null,
        "title": "Alp-Pflege",
      },
    ]
  }
  refine={[Function]}
>
  <SearchBox
    currentRefinement="alp"
    onBlur={[Function]}
    onFocus={[Function]}
    ref={
      Object {
        "current": Object {
          "autocompleteSearch": [Function],
        },
      }
    }
    refine={[Function]}
  >
    <div
      className="search-field"
    >
      <input
        autoFocus={true}
        onKeyUp={[Function]}
        onfocusin={[Function]}
        onfocusout={[Function]}
        oninput={[Function]}
        placeholder="string"
        type="search"
        value="alp"
      />
      <button
        className="search-button"
        onClick={[Function]}
      >
        <img
          alt="string"
          className="w-100 h-100"
          src="test-file-stub"
        />
      </button>
    </div>
  </SearchBox>
  <div
    className="autocomplete-container d-none"
  >
    <div
      className="autocomplete-content"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      ontouchend={[Function]}
      ontouchstart={[Function]}
    >
      <ul>
        <a
          className="autocomplete-hit"
          href="#"
          key="3"
          onClick={[Function]}
        >
          <li
            onMouseOver={[Function]}
          >
            <FontAwesomeIcon
              className="mr-2"
              icon={
                Array [
                  "fas",
                  "search",
                ]
              }
            />
            Alp-Pflege
          </li>
        </a>
      </ul>
    </div>
  </div>
</AutocompleteImpl>
`;

exports[`CustomAutocomplete renders correctly 1`] = `
<AutocompleteImpl
  createURL={[Function]}
  currentRefinement="alp"
  defaultRefinement="alp"
  hits={
    Array [
      Object {
        "__position": 1,
        "_highlightResult": Object {
          "organization_display_name": Object {
            "fullyHighlighted": false,
            "matchLevel": "full",
            "matchedWords": Array [
              "al",
            ],
            "value": "<ais-highlight-0000000000>Al</ais-highlight-0000000000>",
          },
          "plain_description": Object {
            "fullyHighlighted": false,
            "matchLevel": "full",
            "matchedWords": Array [
              "al",
            ],
            "value": "100% Dauer <ais-highlight-0000000000>Al</ais-highlight-0000000000>psommer",
          },
          "title": Object {
            "fullyHighlighted": false,
            "matchLevel": "full",
            "matchedWords": Array [
              "al",
            ],
            "value": "<ais-highlight-0000000000>Al</ais-highlight-0000000000>p-Pflege",
          },
        },
        "brief_description": "Bereich Kulturlandschaft, Biodiversität und Landschaftsqualität",
        "canton_display_name": "Obwalden",
        "category_display_name": "Landwirtschaft",
        "contact_information": "Dieser Betrieb ist noch nicht bei MyZivi registriert. Bitte bewerbe Dich im EZIVI.",
        "description": "MyDescription <b>Using HTML</b>",
        "featured_as_new": true,
        "icon_url": "/myzivi-logo.jpg",
        "identification_number": 89372,
        "language": "german",
        "link": "http://localhost:3000/job_postings/3",
        "minimum_service_months": 1,
        "objectID": "3",
        "organization_display_name": "Alpbetrieb Untersteiglen-Schwand im Tal-Melchsee-Frutt",
        "plain_description": "100% Dauer Alpsommer: von 01.05. bis 15.10",
        "preferred_skills": null,
        "priority_program": true,
        "publication_date": "2021-06-14",
        "published": true,
        "relevancy": 1,
        "required_skills": "<p> Motivation und physische Kondition für harte körperliche Arbeit im Freien</p>",
        "sub_category_display_name": null,
        "title": "Alp-Pflege",
      },
    ]
  }
  refine={[Function]}
>
  <SearchBox
    currentRefinement="alp"
    onBlur={[Function]}
    onFocus={[Function]}
    ref={
      Object {
        "current": Object {
          "autocompleteSearch": [Function],
        },
      }
    }
    refine={[Function]}
  >
    <div
      className="search-field"
    >
      <input
        autoFocus={true}
        onKeyUp={[Function]}
        onfocusin={[Function]}
        onfocusout={[Function]}
        oninput={[Function]}
        placeholder="string"
        type="search"
        value="alp"
      />
      <button
        className="search-button"
        onClick={[Function]}
      >
        <img
          alt="string"
          className="w-100 h-100"
          src="test-file-stub"
        />
      </button>
    </div>
  </SearchBox>
  <div
    className="autocomplete-container d-none"
  >
    <div
      className="autocomplete-content"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      ontouchend={[Function]}
      ontouchstart={[Function]}
    >
      <ul>
        <a
          className="autocomplete-hit"
          href="#"
          key="3"
          onClick={[Function]}
        >
          <li
            onMouseOver={[Function]}
          >
            <FontAwesomeIcon
              className="mr-2"
              icon={
                Array [
                  "fas",
                  "search",
                ]
              }
            />
            Alp-Pflege
          </li>
        </a>
      </ul>
    </div>
  </div>
</AutocompleteImpl>
`;
